File: csharp-selenium-nunit-reqnroll/SauceDemo.UiTests/Pages/CartPage.cs
using OpenQA.Selenium;

namespace SauceDemo.UiTests.Pages;

public sealed class CartPage : BasePage
{
    private static readonly By CartItemNames = By.CssSelector("[data-test='inventory-item-name']");
    private static readonly By CheckoutButton = By.CssSelector("[data-test='checkout']");

    public CartPage(IWebDriver driver) : base(driver)
    {
    }

    public IReadOnlyList<string> ListItemNames() =>
        Driver.FindElements(CartItemNames).Select(e => e.Text).ToList();

    public void RemoveItem(string productName) => Click(RemoveButtonFor(productName));

    public CheckoutInformationPage StartCheckout()
    {
        Click(CheckoutButton);
        return new CheckoutInformationPage(Driver);
    }

    // See InventoryPage.AddToCartButtonFor for why this is XPath-by-class-and-text, not data-test.
    private static By RemoveButtonFor(string productName) => By.XPath(
        $"//div[@class='cart_item'][.//div[@data-test='inventory-item-name' and text()='{productName}']]//button[text()='Remove']");
}

File: python-selenium-pytest-bdd/src/saucedemo_uitests/pages/cart_page.py
from selenium.webdriver.common.by import By

from saucedemo_uitests.pages.base_page import BasePage, Locator
from saucedemo_uitests.pages.checkout_information_page import CheckoutInformationPage

_CART_ITEM_NAMES: Locator = (By.CSS_SELECTOR, "[data-test='inventory-item-name']")
_CHECKOUT_BUTTON: Locator = (By.CSS_SELECTOR, "[data-test='checkout']")


class CartPage(BasePage):
    def list_item_names(self) -> list[str]:
        return [element.text for element in self.driver.find_elements(*_CART_ITEM_NAMES)]

    def start_checkout(self) -> CheckoutInformationPage:
        self.click(_CHECKOUT_BUTTON)
        return CheckoutInformationPage(self.driver)
